app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt

```diff
@@ -98,17 +102,23 @@ class ItemListPresenter(
             .addTo(compositeDisposable)
 
         view.sortItemSelection
-                .subscribe { sortBy ->
-                    dispatcher.dispatch(SettingAction.ItemListSortOrder(sortBy))
-                }.addTo(compositeDisposable)
+            .subscribe { sortBy ->
+                dispatcher.dispatch(SettingAction.ItemListSortOrder(sortBy))
+            }.addTo(compositeDisposable)
 
         accountStore.profile
             .map {
-                (it.value?.displayName ?: it.value?.email).asOptional()
+                (it.value?.asOptional())
             }
-            .filterNotNull()
             .observeOn(AndroidSchedulers.mainThread())
-            .subscribe(view::setDisplayName)
+            .subscribe { profile ->
+                val account = AccountViewModel(
+                    accountName = profile?.value?.displayName ?: profile?.value?.email!!,
+                    displayEmailName = profile?.value?.email!!,
+                    avatarFromURL = profile.value.avatar!!
+                )
+                view.updateAccountProfile(account)
+            }
```

Comment:
This is so much nicer.

From a style pov, I'm leaning towards a `map`/`filterNotNull`/`subscribe` pipeline.

e.g. 

```kotlin
.map { profile -> 
     profile?.value?.apply {
         AccountModel(displayName ?: email, email, avatar)
     }.asOptional()
}
.filterNotNull()
.subscribe(view::updateAccountProfile)
```

Reply:
done